extensions/amp-form/0.1/amp-form.js

```diff
@@ -951,24 +974,33 @@ export class AmpForm {
     }
     return promise.then(responseJson => {
       this.triggerFormSubmitInAnalytics_('amp-form-submit-error');
-      this.handleSubmitFailure_(e, responseJson);
+      this.handleSubmitFailure_(
+        e,
+        responseJson,
+        /** getSsrEventResponse */ false
+      );
       this.maybeHandleRedirect_(e.response);
     });
   }
 
   /**
    * Transition the form the the submit error state.
+   * If getSsrEventResponse is true, exposes the response.body in
+   * the SUBMIT_ERROR event; otherwise exposes the provided json.
    * @param {*} error
    * @param {!JsonObject} json
+   * @param {boolean} getSsrEventResponse
    * @return {!Promise}
    * @private
    */
-  handleSubmitFailure_(error, json) {
+  handleSubmitFailure_(error, json, getSsrEventResponse) {
```

Comment:
Let's avoid this flag argument, since it can [increase complexity and reduce readability](https://martinfowler.com/bliki/FlagArgument.html) and in this case it's making you dupe implementations.

Instead, take a `reduceEventResponseFn` function that's assigned to identity by default. Callers that would set `getSsrEventResponse` to `true`, would instead pass a reducer that returns the body.

```js
const bodyFromSsrEventResponse = json => json['body'];

// ...

handleSubmitSuccess_(jsonPromise, reduceEventResponseFn = json => json) {
  return jsonPromise.then(
    json => {
      const eventResponse = reduceResponseFn(json);
      // ...
    },
    // ...
  );
}


// ...

this.handleSubmitFailure_(error, detail)

// ...

this.handleSubmitSuccess_(tryResolve(() => response), bodyFromSsrEventResponse);
```